### ISSUE TITLE
google_binary_authorization_attestor should accept RSA_SIGN_PSS_4096_SHA256 returned by data.google_kms_crypto_key_version

### ISSUE THREAD

**Issue by kristijan-deanovic-db** [hashicorp/terraform-provider-google#10575]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version
v0.14.11

### Affected Resource(s)

* google_binary_authorization_attestor

### Terraform Configuration Files

```tf
resource "google_binary_authorization_attestor" "built_by_pipeline" {
  name    = "built-by-pipeline"
  project = local.sensitive_a_outputs.cluster_project.project_id

  attestation_authority_note {
    note_reference = google_container_analysis_note.built_by_pipeline.name
    public_keys {
      id = data.google_kms_crypto_key_version.built_by_pipeline.id
      pkix_public_key {
        public_key_pem      = data.google_kms_crypto_key_version.built_by_pipeline.public_key[0].pem
        signature_algorithm = data.google_kms_crypto_key_version.built_by_pipeline.public_key[0].algorithm #returns RSA_SIGN_PSS_4096_SHA256
      }
    }
  }
}
```

### Panic Output

```
Error: Error updating Attestor "projects/db-dev-regt-olbcs-cluster/attestors/built-by-pipeline": googleapi: Error 400: Invalid value at 'attestor.user_owned_grafeas_note.public_keys[0].pkix_public_key.signature_algorithm' (type.googleapis.com/google.cloud.binaryauthorization.v1.PkixPublicKey.SignatureAlgorithm), "RSA_SIGN_PSS_4096_SHA256"
Details:
[
  {
    "@type": "type.googleapis.com/google.rpc.BadRequest",
    "fieldViolations": [
      {
        "description": "Invalid value at 'attestor.user_owned_grafeas_note.public_keys[0].pkix_public_key.signature_algorithm' (type.googleapis.com/google.cloud.binaryauthorization.v1.PkixPublicKey.SignatureAlgorithm), \"RSA_SIGN_PSS_4096_SHA256\"",
        "field": "attestor.user_owned_grafeas_note.public_keys[0].pkix_public_key.signature_algorithm"
      }
    ]
  }
]
```

### Expected Behavior

The algorithm RSA_SIGN_PSS_4096_SHA256 should be a valid value for the resource google_binary_authorization_attestor.

If the data resource google_kms_crypto_key_version returns "RSA_SIGN_PSS_4096_SHA256" as value the attestor resource should accept it and map it to RSA_PSS_4096_SHA256 which is allowed for google_binary_authorization_attestor.

### Actual Behavior

The resource errors with a bad request during apply.

### Steps to Reproduce

1. `terraform apply`

### References

* https://cloud.google.com/nodejs/docs/reference/binary-authorization/latest/binary-authorization/protos.google.cloud.binaryauthorization.v1.pkixpublickey.signaturealgorithm
* https://cloud.google.com/kms/docs/reference/rest/v1/CryptoKeyVersionAlgorithm

**Comment by edwardmedia** [comment]
@kristijan-deanovic-db the error is from api, and the message is clear. Without knowing how you created your key & keyring, it is difficult to say. Are you able to run the example from [here](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/binary_authorization_attestor#example-usage---binary-authorization-attestor-kms)? You can compare the results to see what is wrong and how to fix it.

**Comment by kristijan-deanovic-db** [comment]
@edwardmedia The example looks the same as our implementation.
If you can reproduce the issue then maybe i wasn't precise enough:

We created a google_kms_crypto_key with the purpose `ASYMMETRIC_SIGN`. This resource takes the algorithms mentioned [here](https://cloud.google.com/kms/docs/reference/rest/v1/CryptoKeyVersionAlgorithm). We are using `RSA_SIGN_PSS_4096_SHA256` algorithm.

Next we are using the previously created crypto-key to configure the google_binary_authorization_attestor, using the data.google_kms_crypto_key_version, as seen in the example: 
`signature_algorithm = data.google_kms_crypto_key_version.version.public_key[0].algorithm`

This (data.google_kms_crypto_key_version.version.public_key[0].algorithm) returns the previously set algorithm in google_kms_crypto_key. In our case `RSA_SIGN_PSS_4096_SHA256`. 

Now we run into the issue:

The google_binary_authorization_attestor doesn't accept `RSA_SIGN_PSS_4096_SHA256` as a valid value as seen by the error and the documentation of the signature_algorithm enum [here](https://cloud.google.com/nodejs/docs/reference/binary-authorization/latest/binary-authorization/protos.google.cloud.binaryauthorization.v1.pkixpublickey.signaturealgorithm). 
Instead it accepts `RSA_PSS_4096_SHA256`.

Notice the missing `SIGN` in the algorithm name. I would expect that all the `RSA_SIGN_` algorithms are acceptable values for the `ASYMMETRIC_SIGN` purpose. As this is what the documentation reflects.

Any additional mapping that is necessary to make this work should be handled by the provider.

**Comment by edwardmedia** [comment]
@kristijan-deanovic-db The error shows the API rejects `RSA_SIGN_PSS_4096_SHA256 `. I don't see `RSA_SIGN_PSS_4096_SHA256` available [here](https://cloud.google.com/nodejs/docs/reference/binary-authorization/latest/binary-authorization/protos.google.cloud.binaryauthorization.v1.pkixpublickey.signaturealgorithm) and [here](https://cloud.google.com/binary-authorization/docs/reference/rest/Shared.Types/SignatureAlgorithm) . Instead, API accepts `RSA_SIGN_PKCS1_4096_SHA512` as an example which is available in both docs. Not sure why you think API should accept `RSA_SIGN_PSS_4096_SHA256`? Anyway, this is the controlled by API. The Terraform provider does not have much to do with it. Does this make sense?

**Comment by kristijan-deanovic-db** [comment]
KSM documentation: https://cloud.google.com/kms/docs/reference/rest/v1/CryptoKeyVersionAlgorithm
referencing `RSA_SIGN_PSS_4096_SHA256`

BinAuthZ documentation: https://cloud.google.com/nodejs/docs/reference/binary-authorization/latest/binary-authorization/protos.google.cloud.binaryauthorization.v1.pkixpublickey.signaturealgorithm

The links you provided both show the accepted BinAuthZ key algorithms.

**Comment by edwardmedia** [comment]
@kristijan-deanovic-db I guess the `SignatureAlgorithm` I referenced [here](https://github.com/hashicorp/terraform-provider-google/issues/10575#issuecomment-972932125) is the subset of one you listed [here](https://github.com/hashicorp/terraform-provider-google/issues/10575#issuecomment-972950212). Anyway, whether one is supported or not, it is up to the `attestor` product. The provider does not have much control. You may reach out to the product team to get the explanation. Closing the issue accordingly. Feel free to reopen if you need more support on this

**Comment by MajorBreakfast** [comment]
Please map KMS's `RSA_SIGN_PSS_4096_SHA256` KMS value to the `RSA_PSS_4096_SHA256` BinAuthz value to make it intuitive to use.

I think I've seen such compatibility mappings done elsewhere in the Google provider in cases where different Google products call the same thing by different names.

**Comment by josef-brandl-db** [comment]
Can you plz reopen?

**Comment by slevenick** [comment]
I agree with edwardmedia's assessment that there really isn't a provider issue going on here. I think the better solution would be for the BinAuth API to accept either `RSA_SIGN_PSS_4096_SHA256` or `RSA_PSS_4096_SHA256` as aliases to the same algorithm. Terraform behaves as a pass-through for most string values like this, and putting the transformation logic in Terraform feels like a hack around an API-side inconsistency

**Comment by slevenick** [comment]
Should be fixed, please open a new issue if this is still causing problems.
